--- a/build.rs
+++ b/build.rs
@@ -51,6 +51,7 @@
         Mips,
         Powerpc,
         Powerpc64,
+        Riscv64,
         S390x,
         Wasm32,
     }
@@ -104,6 +105,7 @@
                 "mips" => TargetArch::Mips,
                 "powerpc" => TargetArch::Powerpc,
                 "powerpc64" => TargetArch::Powerpc64,
+                "riscv64" => TargetArch::Riscv64,
                 "s390x" => TargetArch::S390x,
                 "wasm32" => TargetArch::Wasm32,
                 _ => panic!("Unknown target arch: {arch}"),
